public/app.js

```diff
@@ -0,0 +1,330 @@
+// AUTH CONTROLLER
+var authController = (function() {
+    return {
+        //Authentication state listener
+        initApp: function() {
+            firebase.auth().onAuthStateChanged(function (user) {
+                if (user) {
+                    console.log('Login Successful');
+                    console.log("Attempting to bind: " + user.email)
+                    console.log('Bind Successful');
+                    return user;
+                } else {
+                    window.location.replace('auth.html')
+                }
+            });
+        },
+
+        //logout function
+        logout: function() {
+            firebase.auth().signOut()
+                .catch(function (err) {
+                    console.log(err);
+                })
+        }
+    };
+})();
+
+// DATA CONTROLLER
+var dataController = (function() {
+    var data = [];
```

Comment:
(note for other reviewers: this block to the end of the updateData function is copied unmodified from master)

Reply:
noted